PropInProgress closer_marker
PropInProgress farther_marker